modules/@apostrophecms/modal/ui/apos/components/AposModalShareDraft.vue

```diff
@@ -79,22 +86,90 @@ export default {
         disableHeader: true,
         trapFocus: true
       },
-      disabled: false
+      shareUrl: '',
+      disabled: true
     };
   },
   async mounted() {
     this.modal.active = true;
+    this.checkUrlprop();
   },
   methods: {
-    copy() {
+    async copy() {
       console.log('copy');
+      await navigator.clipboard.writeText(this.shareUrl);
     },
-    toggle() {
-      console.log('toggle');
+    async toggle() {
       this.disabled = !this.disabled;
+
+      await this.setShareUrl();
+    },
+    async setShareUrl() {
+      try {
+        const { aposShareKey } = await apos.http.post(
+          `${apos.modules[this.doc.type].action}/${this.doc._id}/share`, {
+            busy: true,
+            body: {
+              share: !this.disabled
+            },
+            draft: true
+          }
+        );
+
+        if (this.disabled) {
+          this.shareUrl = '';
+          return;
+        }
+
+        if (!aposShareKey) {
+          return this.errorNotif();
+        }
+
+        this.shareUrl = this.generateShareUrl(aposShareKey);
+
+      } catch (err) {
+        if (!this.disabled) {
+          this.errorNotif();
+        } else {
+          this.shareUrl = '';
+        }
```

Comment:
I'd switch the conditions for a better readability:

```suggestion
        if (this.disabled) {
          this.shareUrl = '';
          return;
        }
        this.errorNotif();
```

🥖

Reply:
👍🏼